db/db_path_supplier.h

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+
+#include "db/column_family.h"
+#include "rocksdb/status.h"
+
+#define UNUSED(expr) do { (void)(expr); } while (0)
+
+namespace rocksdb {
+
+// An object vendored by column family to dynamically supply db path to
+// functions that need to decide which db_path to flush an sst file to. The
+// supplier object is mutable (because you can add file size to it) and
+// can update global file counters, so usage should be inside proper locking.
+class DbPathSupplier {
+ public:
+  explicit DbPathSupplier(const ImmutableCFOptions& ioptions):
+    ioptions_(ioptions) {}
+
+  virtual ~DbPathSupplier() = default;
+
+  Status FsyncDbPath(uint32_t path_id) const;
+
+  virtual uint32_t GetPathId(int level) const {
+    UNUSED(level);
```

Comment:
The practice followed in this code base is to declare it as `int /* level */`

Reply:
Fixed